From 7d90fa97dd78819865027bb0bceff4c04df06a15 Mon Sep 17 00:00:00 2001
From: Scott K Logan <logans@cottsay.net>
Date: Sun, 23 Mar 2014 20:23:51 -0600
Subject: [PATCH] Better detection of xpath usage

The version of xpath on Fedora functions like the version that Apple has. Instead of making the syntax different only for Apple, parse the xpath usage.
---
 UseOROCOS-RTT-helpers.cmake | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/UseOROCOS-RTT-helpers.cmake b/UseOROCOS-RTT-helpers.cmake
index 35c803f..ac82b4c 100644
--- UseOROCOS-RTT-helpers.cmake.orig	2014-07-03 16:11:09.762621000 +0200
+++ UseOROCOS-RTT-helpers.cmake	2014-07-03 16:12:10.997714000 +0200
@@ -14,13 +14,14 @@
   if (NOT XPATH_EXE)
     message("Warning: xpath not found. Can't read dependencies in manifest.xml file.")
   else(NOT XPATH_EXE)
-    IF (APPLE)
+    execute_process(COMMAND ${XPATH_EXE} ERROR_VARIABLE XPATH_USAGE)
+    IF (NOT XPATH_USAGE MATCHES ".*(-e query).*")
       execute_process(COMMAND ${XPATH_EXE} ${CMAKE_CURRENT_SOURCE_DIR}/manifest.xml "package/depend/@package" RESULT_VARIABLE RES OUTPUT_VARIABLE DEPS)
       SET(REGEX_STR " package=\"([^\"]+)\"")
-    ELSE (APPLE)
+    ELSE (NOT XPATH_USAGE MATCHES ".*(-e query).*")
       execute_process(COMMAND ${XPATH_EXE} -q -e "package/depend/@package" ${CMAKE_CURRENT_SOURCE_DIR}/manifest.xml RESULT_VARIABLE RES OUTPUT_VARIABLE DEPS)
       SET(REGEX_STR " package=\"([^\"]+)\"\n")
-    ENDIF (APPLE)
+    ENDIF(NOT XPATH_USAGE MATCHES ".*(-e query).*")
     if (NOT RES EQUAL 0)
       message(SEND_ERROR "Error: xpath found but returned non-zero:${DEPS}")
     endif (NOT RES EQUAL 0)
